tests/aws/services/stepfunctions/v2/test_sfn_api.py

```diff
@@ -491,6 +493,21 @@ def test_stop_execution(
         sfn_snapshot.match("exec_resp", exec_resp)
         execution_arn = exec_resp["executionArn"]
 
+        def _check_stated_entered(events: HistoryEventList) -> bool:
+            # Check the evaluation entered the wait state, called State_1.
+            for event in events:
+                event_details = event.get("stateEnteredEventDetails")
+                if event_details:
+                    return event_details.get("name") == "State_1"
+            return False
+
+        # Wait until the state machine enters the wait state.
+        await_on_execution_events(
+            stepfunctions_client=aws_client.stepfunctions,
+            execution_arn=execution_arn,
+            check_func=_check_stated_entered,
+        )
+
         await_execution_started(
```

Comment:
This `await_execution_started` waiter seems redundant now because once we entered the waiting state `State_1`, the execution must have started, right?
I guess it does not hurt and simply adds a minor delay but it does not seem necessary from a test logic perspective and could be confusing. Looking at the snapshot of `exec_hist_resp`:
1. `executionStartedEventDetails`
2. `stateEnteredEventDetails` with the name `State_1` (i.e., the waiting state)

Reply:
Thank you @joe4dev, indeed. Unfortunately I merged this seconds before the review came through. I will address this in a following PR, as I want to address another flaky test I have found!

Reply:
No worries. It's more important to get the flake fixed quickly 👍  All good 😃